### ISSUE TITLE
[RenovateBot] - Ensure ArgoCD Chart + .tool-version CLI is bumped in parallel by the same PR

### ISSUE THREAD

**Issue by julianxhokaxhiu** [nearform/k8s-kurated-addons#24]
### Issue Type

Enhancement

### Description

**The Context:**
In its current state, this project uses the [Renovate Bot](https://github.com/renovatebot/renovate) app to bump dependencies and update them automatically without human intervention. Although this is useful for long-term maintenance, there are some tweaks required to ensure the ArgoCD dependency is bumped correctly.

**The Solution:**
We need a special configuration that handles BOTH these cases and ensures they come up under one single PR somehow:
- Bump the [ArgoCD chart version](https://github.com/nearform/k8s-kurated-addons/blob/main/addons/argocd/Chart.yaml#L7) which is not the same as ArgoCD itself AND,
- Bump the [CLI version](https://github.com/nearform/k8s-kurated-addons/blob/main/.tool-versions#L1) following the ArgoCD version [defined in the Chart](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml#L2)

The main challenge here is to ensure two different managers cooperate ( [asdf](https://docs.renovatebot.com/modules/manager/asdf/) for `.tool-versions` and [helmv3](https://docs.renovatebot.com/modules/manager/helmv3/) for Charts ) accordingly to ensure they both produce a commit under a single PR. It is totally fine to have two different commits under the same PR as we anyway use the [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) strategy.

**The Alternative:**
None is considered, but any idea is welcome.

### Detailed steps

```bash
N/A
```

**Comment by julianxhokaxhiu** [comment]
Since the last week, I managed to accomplish the following things:
- Added the `argocd` plugin detection on the asdf manager on the renovate bot mainline code ( https://github.com/renovatebot/renovate/pull/19941 )
- Created a simple repro case to test the aforementioned implementation and to open the door for discussing how to tackle this issue, on a simpler repo ( https://github.com/julianxhokaxhiu/renovate-test )
- Discussed the possibility of having this feature ( or how to achieve it ) on the official renovate discussion board ( https://github.com/renovatebot/renovate/discussions/19929 )

After all the discussion unfortunately it seems the renovate bot doesn't support this scenario and the team is not really interested in adding this feature ( for now ).

Therefore, after an internal alignment, there were some ideas that were brought to the table. The main idea is to trigger an action workflow that would trigger when either the [`.tool-versions`](https://github.com/nearform/k8s-kurated-addons/blob/main/.tool-versions) or the [argocd Chart](https://github.com/nearform/k8s-kurated-addons/tree/main/addons/argocd) has been changed, but different scenarios could be triggered:
- ( more feasible ) Cross-check that they are both pointing to the same argocd version. If not, fail the CI job, if yes, mark it as successful OR;
- (less feasible) When one of the two is updated, bump the other one

As this is not a high priority per se, we will put it on hold and wait until this issue really happens. If it will, we could pick it up and implement the aforementioned actions.
